clients/include/testing_spr2.hpp

```diff
@@ -0,0 +1,200 @@
+/* ************************************************************************
+ * Copyright 2018-2020 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "flops.hpp"
+#include "near.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T>
+void testing_spr2_bad_arg()
+{
+    rocblas_fill         uplo  = rocblas_fill_upper;
+    rocblas_int          N     = 100;
+    rocblas_int          incx  = 1;
+    rocblas_int          incy  = 1;
+    T                    alpha = 0.6;
+    rocblas_local_handle handle;
+
+    size_t abs_incx = incx >= 0 ? incx : -incx;
+    size_t abs_incy = incy >= 0 ? incy : -incy;
+    size_t size_A   = size_t(N) * (N + 1) / 2;
+    size_t size_x   = size_t(N) * abs_incx;
+    size_t size_y   = size_t(N) * abs_incy;
+
+    // allocate memory on device
+    device_vector<T> dA_1(size_A);
+    device_vector<T> dx(size_x);
+    device_vector<T> dy(size_y);
+    CHECK_HIP_ERROR(dA_1.memcheck());
+    CHECK_HIP_ERROR(dx.memcheck());
+    CHECK_HIP_ERROR(dy.memcheck());
+
+    EXPECT_ROCBLAS_STATUS(rocblas_spr2<T>(handle, uplo, N, &alpha, nullptr, incx, dy, incy, dA_1),
+                          rocblas_status_invalid_pointer);
```

Comment:
Can pass uplo == rocblas_fill_full and check for invalid_value return?   I'm doing that in all new functions

Reply:
Yup, I'll add that.